### PR TITLE
Support scalar in `chainerx::Where`

### PR DESCRIPTION
**Pull request by asi1024**

### CODE REVIEW

File: chainerx_cc/chainerx/cuda/cuda_device/indexing.cu [hunk 1]
```diff
@@ -294,6 +294,88 @@ public:
...
+            using CudaType = cuda_internal::DataType<T>;
+            Elementwise<const bool, const T, T>(WhereASImpl<T>{static_cast<CudaType>(y)}, condition_cast, x_cast, out);
+        });
+    }
+};
+
+CHAINERX_CUDA_REGISTER_KERNEL(WhereASKernel, CudaWhereASKernel);
+
+template <typename T>
+struct WhereSAImpl {
```
Comment (niboshi) [inline chainerx_cc/chainerx/cuda/cuda_device/indexing.cu]: `WhereASAImpl`?

File: chainerx_cc/chainerx/cuda/cuda_device/indexing.cu [hunk 2]
```diff
@@ -294,6 +294,88 @@ public:
...
+            using CudaType = cuda_internal::DataType<T>;
+            Elementwise<const bool, const T, T>(WhereSAImpl<T>{static_cast<CudaType>(x)}, condition_cast, y_cast, out);
+        });
+    }
+};
+
+CHAINERX_CUDA_REGISTER_KERNEL(WhereSAKernel, CudaWhereSAKernel);
+
+template <typename T>
+struct WhereSSImpl {
```
Comment (niboshi) [inline chainerx_cc/chainerx/cuda/cuda_device/indexing.cu]: `WhereASSImpl`?

File: chainerx_cc/chainerx/cuda/cuda_device/indexing.cu [hunk 3]
```diff
@@ -294,6 +294,88 @@ public:
 
 CHAINERX_CUDA_REGISTER_KERNEL(WhereKernel, CudaWhereKernel);
 
+template <typename T>
+struct WhereASImpl {
```
[the remark this answers is not in the document]
Comment (asi1024) [reply chainerx_cc/chainerx/cuda/cuda_device/indexing.cu]: Fixed.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Please fix the test.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
LGTM

**Pull request merged**